subsys/bluetooth/host/att.c

```diff
@@ -1548,6 +1548,19 @@ static uint8_t att_read_type_req(struct bt_att_chan *chan, struct net_buf *buf)
 		return 0;
 	}
 
+	/* Reading Database Hash is special as it is required to make client
```

Comment:
In all the other cases we have this part

```c
	if (!bt_gatt_change_aware(chan->att->conn, true)) {
		if (!atomic_test_and_set_bit(chan->flags, ATT_OUT_OF_SYNC_SENT)) {
			return BT_ATT_ERR_DB_OUT_OF_SYNC;
		} else {
			return 0;
		}
	}
```

in the `rsp` functions. 

To be consistent, shouldn't we simply add the above to `att_read_type_rsp`?

Reply:
If the point of this PR is adding the additional check `if (bt_uuid_cmp(&u.uuid, BT_UUID_GATT_DB_HASH) != 0) {`, then shouldn't that check be added to all read handlers? What if the client reads the database with `BT_ATT_OP_READ_REQ` instead of `BT_ATT_OP_READ_TYPE_REQ`?

Reply:
This is not very consistent as some (att_read_mult_req, att_read_mult_vl_req) are handled in req handlers,  but I can do this in rsp too

Reply:
> This is not very consistent as some (att_read_mult_req, att_read_mult_vl_req) are handled in req handlers, but I can do this in rsp too

Planning any updates on this? 
It's unclear to me whether this PR needs a change or not